src/main/scala/org/ergoplatform/ErgoConstants.scala

```diff
@@ -0,0 +1,67 @@
+package org.ergoplatform
+
+import org.ergoplatform.ErgoLikeContext
+import sigmastate.{AtLeast, SBigInt, SPrimType}
+
+case class SizeConstant[T: Numeric](value: T, id: Short, description: String) {
+  def get: T = value
+}
+
+/**
+  * Fundamental constants that are used in sigma's logic and checks
+  */
+object ErgoConstants {
+
+  object MaxInputSize extends SizeConstant[Int](1024 * 1024 * 1, 1,
+    "Input size should not be greater then provided value") {
+  }
+
+  object MaxTreeDepth extends SizeConstant[Int](110, 2,
```

Comment:
While 110 is probably OK, a more conservative and maybe safer option is to lower it to e.g. to 64.

Reply:
in this case a test `big byte array with a lot of operations` from `SpamSpecification` fails, probably we need to change the test then

Reply:
fixed